docs/devel/offline-virtual-machine.md

```diff
@@ -0,0 +1,291 @@
+is used. As described in the
+[design document](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/api-machinery/controller-ref.md),
+the OwnerReference lives in the metadata section of the object and is created
+automaticaly. Example:
+
+```YAML
+apiVersion: kubevirt.io/v1alpha1
+kind: VirtualMachine
+metadata:
+  name: myvm
+  ownerReferences:
+    - controller: true
+      uid: 1234-1234-1234-1234
+      kind: OfflineVirtualMachine
+      version: kubevirt.io/v1alpha1
+```
+
+### Update strategy
+
+For now implicit: OnDelete. Can later be extended to RollingUpdate if needed.
+Spec changes have no direct effect on already running VMs, and they will not
+directly be propagated to the VM. If a VM should be running (spec.running=true)
+and it is powered down (VM object delete, OS shutdown, ...),
+the VM will be re-created by the controller with the new spec.
+
+### Delete strategy
+
+The delete has a cascade that deletes the created VirtualMachine. If a cascade
+is turned off the VirtualMachine is orphaned and leaved running.
+When the OfflineVirtualMachine with the same name as orphaned VirtualMachine
+is created, the VirtualMachine gets adopted and OwnerReference
+is updated accordingly.
+
+### Reset and Reboot
+
+This is not covered by the OfflineVirtualMachine. This functionality shall
+be achieved by subresources for the VirtualMachine (imperative),
+and will not result in a recreation of the VirtualMachine object or its Pod.
+From the KubeVirt perspective, the VirtualMachine is running all the time.
+Thus spec changes on OfflineVirtualMachine will not be propagated to
+the VirtualMachine in that case.
+
+## How it is implemented
+
+The first implementation of the OfflineVirtualMachine kind has two parts:
+
+1) The OfflineVirtualMachine custom resource definition,
+2) The controller watching for the changes and updating the state.
+
+The OfflineVirtualMachine custom resource is straightforward and is shown bellow:
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: offlinevirtualmachines.kubevirt.io
+spec:
+  scope: Namespaced
+  group: kubevirt.io
+  version: v1alpha1
+  names:
+    kind: OfflineVirtualMachine
+    plural: offlinevirtualmachines
+    singular: offlinevirtualmachine
+    shortNames:
+    - ovm
+    - ovms
+```
+
+Part of the definition of custom resource is a API specification used for
+autogenerating the Kubernetes resources: client, lister and watcher.
+
+The controller is responsible for watching the change in the registered
+offline virtual machines and update the state of the system. It is also
+responsible for creating new VirtualMachine when the `running` is set to `true`.
+Moreover the controller attaches the `metadata.OwnerReference` to the created
+VirtualMachine. With this mechanism it can link the OfflineVirtualMachine to the
+VirtualMachine and show combined status.
```

Comment:
Here and further above you speak about updating the status.

It would be helpful if you are more precise what part of the status will be updated and how. An example would be helpful.

Reply:
Covered now in https://github.com/kubevirt/kubevirt/pull/641/files#diff-118110639d0782e15a8dc5508de66823R201